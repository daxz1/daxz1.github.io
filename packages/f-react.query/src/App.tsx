import {QueryClient, QueryClientProvider} from "react-query";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core";
import { createMuiTheme } from "@material-ui/core/styles";
import { ReactQueryDevtools } from "react-query/devtools";
import Layout from "./components/Layout";

const queryClient = new QueryClient();

export default function App(): JSX.Element {
    return (
        <QueryClientProvider client={queryClient}>
            <Router>
                <ThemeProvider theme={theme} >
                    <Layout />
                    <ReactQueryDevtools initialIsOpen />
                </ThemeProvider>
            </Router>
        </QueryClientProvider>
    );
}

const theme = createMuiTheme({
    typography: {
        h1: {
            fontFamily: "Roboto Mono, monospace",
        },
        h2: {
            fontFamily: "Roboto Mono, monospace",
        },
        h3: {
            fontFamily: "Roboto Mono, monospace",
        },
        h4: {
            fontFamily: "Roboto Mono, monospace",
        },
        h5: {
            fontFamily: "Roboto Mono, monospace",
        },
        h6: {
            fontFamily: "Roboto Mono, monospace",
        },
    },
});
