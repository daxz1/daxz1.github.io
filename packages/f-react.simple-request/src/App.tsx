import { useState, useEffect } from 'react';

function sleep(ms:number):Promise<any> {
  return new Promise(resolve => setTimeout(resolve, ms));
}

interface ICharacter {
  created: string;
  episode: [],
  gender: string;
  image: string;
  location: [];
  origin: [];
  url: string;
  id: string;
  name: string;
  status: string;
  species: string;
  type: string;
}

const fetchCharacter = async (id: string | undefined): Promise<ICharacter> => {
  await sleep(5000);
  if (id === undefined) {
    throw new Error('id undefined');
  }
  const response = await fetch(`https://rickandmortyapi.com/api/character/${id}`);
  if (!response.ok) {
    throw new Error('Fetch Error');
  }
  return await response.json();
};


export default function App ():JSX.Element {

  const [character, setCharacter] = useState(undefined as ICharacter | undefined);

  useEffect(()=> {
    (async () => {
      try {
        const response = await fetchCharacter('1');
        setCharacter(response);
      } catch (e) {
        // Set an
      }
    })();
  })

  return (
    <div>
      {character?.id}
      <br/>
      {character?.name}
    </div>
  );
}
