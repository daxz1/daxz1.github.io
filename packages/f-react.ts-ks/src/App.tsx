import React from 'react';
import TSComponent from './Components/TSComponent';
import JSComponent from './Components/JSComponent';

const App:React.FC = () => {
  return (
    <>
      <TSComponent name="Foo"/>
      <JSComponent name="bar" />
    </>
  );
}

export default App;
