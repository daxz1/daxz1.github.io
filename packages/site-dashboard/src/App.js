import React from 'react';
import {DailyPositive, TotalPositive, DailyTestsProcessed, TotalTestsProcessed} from "./containers";
import {Layout, TileWithCountUp } from "./components";

export const App = () => {

  return (
    <Layout>
      <div className="row_one">
        <DailyPositive/>
        <TotalPositive/>
        <DailyTestsProcessed/>
        <TotalTestsProcessed/>
      </div>
    </Layout>
  );
}

