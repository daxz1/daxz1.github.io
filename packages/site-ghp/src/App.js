import React from 'react';
import { SocialIcon } from 'react-social-icons';
import portrait from './images/portrait.jpg';

const SOCIAL_URLS = [
  'https://www.twitter.com/davsinghc',
  'https://www.linkedin.com/in/dxc/',
];

function App() {
  return (
    <>
    <section className="flex__container">
      <div className="grid__container">
        <main className="content">
          <div className="portrait--container">
            <img src={portrait} alt="Dav Singh (daxz1)" className="portrait"/>
          </div>
          <p className="text__align--center">daxz1</p>
          <p className="text__align--center">
            Frontend Engineer currently working at {' '}
            <a href="https://www.prodyna.com/en/" target="_blank" rel="noopener noreferrer" className="text__link--no-styles"><strong>@prodyna</strong></a> {' '}
            London office.
          </p>
          <div className="flex__container">
            <div className="social__container">
              { SOCIAL_URLS.map(url => {
                return (<SocialIcon
                  key={url}
                  url={url}
                  bgColor="rgb(255,255,255)"
                  style={{ height: 25, width: 25 }}
                />)
              })}
            </div>
          </div>
        </main>
      </div>
    </section>

    <section className='flex__container'>
      <div className='grid__container--full'>
        <div className='content section__projects'>
          <div className="powwownow">
            Powwownow
          </div>
          <div className='fn'>
            Financial News
          </div>
        </div>
      </div>
    </section>
    </>
  );
}

export default App;
