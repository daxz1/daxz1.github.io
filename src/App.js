import React from 'react';

function App() {
  return (
    <div>Coming Soon</div>
  );
}

export default App;
