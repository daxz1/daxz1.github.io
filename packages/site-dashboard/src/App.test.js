import React from "react";
import { create } from "react-test-renderer";
import { App } from "./App";

// Dummy test
describe("Application component", () => {
  test("Matches the snapshot", () => {
    const Application = create(<App />);
    expect(Application.toJSON()).toMatchSnapshot();
  });
});
