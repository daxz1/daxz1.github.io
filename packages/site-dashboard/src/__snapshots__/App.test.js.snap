// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application component Matches the snapshot 1`] = `
<div
  className="wrapper bg-colour bg-colour--woodsmoke"
>
  <div
    className="grid"
  >
    <div
      className="row_one"
    >
      <div
        className="tile"
      >
        <div
          className="loader"
        />
      </div>
      <div
        className="tile"
      >
        <div
          className="loader"
        />
      </div>
      <div
        className="tile"
      >
        <div
          className="loader"
        />
      </div>
      <div
        className="tile"
      >
        <div
          className="loader"
        />
      </div>
    </div>
  </div>
</div>
`;
